handlers/holiday-stop-processor/src/main/scala/com/gu/holidaystopprocessor/HolidayStopProcess.scala

```
@@ -1,18 +1,46 @@
+import com.gu.salesforce.holiday_stops.SalesforceHolidayStopRequestActionedZuoraRef.{HolidayStopRequestActionedZuoraAmendmentCode, HolidayStopRequestActionedZuoraAmendmentPrice}
+
 object HolidayStopProcess {
 
   def apply(config: Config): Seq[Either[String, HolidayStopResponse]] = {
+    val sfCredentials = config.sfCredentials
     val zuoraCredentials = config.zuoraCredentials
-    val response = processHolidayStop(
+    processHolidayStops(
       config,
+      getRequests = Salesforce.holidayStopRequests(sfCredentials),
       getSubscription = Zuora.subscriptionGetResponse(zuoraCredentials),
       updateSubscription = Zuora.subscriptionUpdateResponse(zuoraCredentials),
-      getLastAmendment = Zuora.lastAmendmentGetResponse(zuoraCredentials)
+      getLastAmendment = Zuora.lastAmendmentGetResponse(zuoraCredentials),
+      exportAmendments = Salesforce.holidayStopUpdateResponse(sfCredentials)
+    )
+  }
+
+  def processHolidayStops(
+    config: Config,
+    getRequests: String => Either[String, Seq[HolidayStopRequest]],
+    getSubscription: String => Either[String, Subscription],
+    updateSubscription: (Subscription, SubscriptionUpdate) => Either[String, Unit],
+    getLastAmendment: Subscription => Either[String, Amendment],
+    exportAmendments: Seq[HolidayStopResponse] => Either[String, Unit]
+  ): Seq[Either[String, HolidayStopResponse]] = {
+    val response = processHolidayStop(
+      config,
+      getSubscription,
+      updateSubscription,
+      getLastAmendment
     ) _
-    HolidayStop.holidayStopsToApply(config) match {
+    HolidayStop.holidayStopsToApply(getRequests) match {
       case Left(msg) => Seq(Left(msg))
-      case Right(holidayStops) => holidayStops map response
+      case Right(holidayStops) =>
+        val responses = holidayStops.map(response)
+        val exportResult = exportAmendments(responses.collect { case Right(successes) => successes })
```

We seem to discard errors in `responses` which are results of applying holiday stop in Zuora.

---

I have a new return type now, which separates out individual exceptions in holiday stops from general processing exceptions.  So there's no loss of information in what's returned.